generators/generator-microsoft-bot-adaptive/generators/app/index.js

```diff
@@ -144,41 +143,77 @@ module.exports = class extends Generator {
     // 8. end - Called last, cleanup, say good bye, etc
 
     writing() {
-        this._copyDotnetProject();
-        this._copySchemas();
-        this._writeNugetConfig();
+        this._copyProject();
 
         if (this.includeApplicationSettings) {
             this._writeApplicationSettings();
         }
     }
 
-    _copyDotnetProject() {
-        const botName = this.options.botName;
-        const integration = this.options.integration;
-        const platform = this.options.platform;
-        const packageReferences = this._formatPackageReferences();
+    _copyProject() {
+        switch (this.options.platform) {
+            case PLATFORM_DOTNET: {
+                this._copyPlatformTemplate({
+                    defaultSettingsDirectory: 'string.Empty',
+                    includeAssets: ['schemas'],
+                    templateContext: {
+                        packageReferences: this._formatDotnetPackageReferences(this.packageReferences),
+                    },
+                });
+
+                this._copyDotnetSolutionFiles();
+                this._writeNugetConfig();
+
+                return;
+            }
+
+            case PLATFORM_JS: {
+                this._copyPlatformTemplate({
+                    defaultSettingsDirectory: 'process.cwd()',
+                    includeAssets: ['schemas'],
+                });
+
+                this._writeJsPackageJson();
+
+                return;
+            }
+        }
+    }
+
+    _copyPlatformTemplate({
```

Comment:
This method is now generic for use on all platforms. Specific functionality was extracted to platform-specific calls.